MAVProxy/mavproxy.py

```diff
@@ -1139,7 +1140,7 @@ def quit_handler(signum = None, frame = None):
     # up on exit
     while (mpstate.status.exit != True):
         try:
-            if opts.daemon:
+            if (opts.daemon == True or opts.non_interactive == True):
```

Comment:
Just asserting rather than checking for equivalence is sufficient here.  Also, Python usually excludes the prens unless required.  So
```if opts.daemon or opts.non_interactive:```